ouroboros-network-protocols/test/Test/AnchoredFragment.hs

```diff
@@ -222,6 +224,14 @@ prop_selectPoints (TestBlockAnchoredFragment c) =
   where
     offsets = [0,1,2,3,5,8,13,21,34,55,89,144,233,377,610,987,1597,2584]
 
+prop_foldr :: TestJoinableAnchoredFragments -> Property
```

Comment:
I wonder if it's possible to test a more fundamental property, namely:

```
AS.foldr f z (mkAnchoredSeq ... xs) == fold f z xs
```

Reply:
Actually this is mimicking the definition of `AS.foldr`. So let's leave the test as is.